hedera-node/src/main/java/com/hedera/services/state/exports/SignedStateBalancesExporter.java

```diff
@@ -165,6 +204,82 @@ private boolean exportBalancesFile(BalancesSummary summary, String csvLoc, Insta
+			AllAccountBalances.Builder allAccountBalancesBuilder) {
+
+		long nodeBalanceWarnThreshold = dynamicProperties.nodeBalanceWarningThreshold();
+		BigInteger totalFloat = BigInteger.valueOf(0L);
+
+		var nodeIds = MiscUtils.getNodeAccounts(signedState.addressBook());
+		var tokens = signedState.tokens();
+		var accounts = signedState.accounts();
+		var tokenAssociations = signedState.tokenAssociations();
+
+		for (MerkleEntityId id : accounts.keySet()) {
+			var account = accounts.get(id);
+			if (!account.isDeleted()) {
+				var accountId = id.toAccountId();
+				var balance = account.getBalance();
+				if (nodeIds.contains(accountId) && balance < nodeBalanceWarnThreshold) {
+					log.warn(String.format(
+							LOW_NODE_BALANCE_WARN_MSG_TPL,
+							readableId(accountId),
+							balance));
+				}
+				totalFloat = totalFloat.add(BigInteger.valueOf(account.getBalance()));
+
+				SingleAccountBalances.Builder singleAccountBuilder =SingleAccountBalances.newBuilder();
+				singleAccountBuilder.setAccountID(accountId)
+						.setHbarBalance(balance);
+
+				if (dynamicProperties.shouldExportTokenBalances()) {
+					var accountTokens = account.tokens();
+					if (accountTokens.numAssociations() > 0) {
+						for (TokenID tokenId : accountTokens.asIds()) {
+							var token = tokens.get(fromTokenId(tokenId));
+							if (token != null && !token.isDeleted()) {
+								var relationship = tokenAssociations.get(fromAccountTokenRel(accountId, tokenId));
+								singleAccountBuilder.addTokenBalances(tb(tokenId, relationship.getBalance()));
+							}
+						}
+					}
+				}
+				Timestamp.Builder consensusTimeStamp = Timestamp.newBuilder();
+				consensusTimeStamp.setSeconds(when.getEpochSecond()).setNanos(when.getNano());
+				allAccountBalancesBuilder.setConsensusTimestamp(consensusTimeStamp.build());
+				allAccountBalancesBuilder.addAllAccounts(singleAccountBuilder.build());
+			}
+		}
+		return totalFloat;
+	}
+
+	private boolean exportBalancesProtoFile(AllAccountBalances.Builder allAccountsBuilder, String protoLoc) {
+		if(log.isDebugEnabled()) {
+			log.debug("Export all accounts to protobuf file {} ", protoLoc);
+		}
+
+		try (FileOutputStream fout = new FileOutputStream(protoLoc)) {
+			allAccountsBuilder.build().writeTo(fout);
+		} catch (IOException e) {
+			log.error(String.format(BAD_EXPORT_ATTEMPT_ERROR_MSG_TPL, protoLoc), e);
+			return false;
+		}
+		return true;
+	}
+
+	public Optional<AllAccountBalances> importBalanceProtoFile(String protoLoc) {
+		try {
+			FileInputStream fin = new FileInputStream(protoLoc);
+			AllAccountBalances allAccountBalances = AllAccountBalances.parseFrom(fin);
+			return Optional.ofNullable(allAccountBalances);
+		} catch (InvalidProtocolBufferException e) {
+			log.error("protobuf message file is corrupted: {}", protoLoc);
+		} catch (IOException e) {
+			log.error("Can't read protobuf message file {}", protoLoc);
+		}
+		return Optional.empty();
```

Comment:
can we add test cases such that, given AllAccountBalances.parseFrom(fin) will throw InvalidProtocolBufferException and IOException and validate we return empty AllAccountBalances

Reply:
Yeah, I will.